packages/expo-navigation-bar/src/NavigationBar.ts

```
@@ -1,133 +1,39 @@
+// doesn't import the actual module in the compiled JavaScript code.  The actual
+// module is imported conditionally with require() below, in order to avoid
+// importing the module if edge-to-edge is not enabled (which could throw if
+// it's not linked).
+let SystemBars: typeof import('react-native-edge-to-edge').SystemBars | null = null;
 
-/**
- * Observe changes to the system navigation bar.
- * Due to platform constraints, this callback will also be triggered when the status bar visibility changes.
- *
- * @example
- * ```ts
- * NavigationBar.addVisibilityListener(({ visibility }) => {
- *   // ...
- * });
- * ```
- */
-export function addVisibilityListener(
-  listener: (event: NavigationBarVisibilityEvent) => void
-): EventSubscription {
-  if (__DEV__) {
-    potentiallyWarnAboutEdgeToEdge();
-  }
-  // Assert so the type is non-nullable.
-  if (!ExpoNavigationBar.addListener) {
-    throw new UnavailabilityError('NavigationBar', 'addVisibilityListener');
-  }
-  return ExpoNavigationBar.addListener('ExpoNavigationBar.didChange', listener);
+if (isEdgeToEdge() && Platform.OS === 'android') {
+  SystemBars = require('react-native-edge-to-edge').SystemBars;
 }
 
+// MARK: react-native-edge-to-edge based APIs
 /**
- * Changes the navigation bar's background color.
- *
- * @example
- * ```ts
- * NavigationBar.setBackgroundColorAsync("white");
- * ```
- * @param color Any valid [CSS 3 (SVG) color](http://www.w3.org/TR/css3-color/#svg-color).
+ * Sets the style of the navigation bar.
+ * > This is only supported on Android when edge-to-edge is enabled.
  */
-export async function setBackgroundColorAsync(color: string): Promise<void> {
-  if (__DEV__) {
-    potentiallyWarnAboutEdgeToEdge();
-  }
-  if (Platform.OS !== 'android') {
-    console.warn('`setBackgroundColorAsync` is only available on Android');
+export function setStyle(style: NavigationBarStyle) {
+  if (!isEdgeToEdge() && Platform.OS === 'android') {
```

```suggestion
  if (!isEdgeToEdgeEnabled) {
    throw new Error('`setStyle` is only supported on Android when edge-to-edge is enabled. Enable edge-to-edge or use the `setButtonStyle` function instead.');
  }
```
I would prefer to throw an error instead of a warning. Additionally, we should consider throwing an error or showing a warning on other platforms as well.